src/System.Device.Gpio/System/Device/Gpio/WaitForEventResult.cs

```diff
@@ -10,10 +10,15 @@ namespace System.Device.Gpio
     public struct WaitForEventResult
     {
         /// <summary>
-        /// The event types to wait for.
+        /// The event types that were waited for.
         /// </summary>
         public PinEventTypes EventTypes;
 
+        /// <summary>
+        /// The event type that was actually seen
+        /// </summary>
+        public PinEventTypes DetectedEventTypes;
```

Comment:
Currently this is supposed to be the EventTypes as this is a result structure. If we want to add public surface area like this, we would require to first log an issue with the proposal so that it could be discussed before we add it to our API.

Reply:
@joperezr See Issue #925: The documentation for `EventTypes` said, quite ambiguously "The event types to wait for.", which can be interpreted as the desired event types for the wait. That's what was returned by the linux implementation (but that's IMHO quite useless). The Windows implementation, on the other hand, returned the actually seen event type. 
So we can either 
- Fix the linux implementation and make the documentation more clear
- Add this extra member and clearly distinguish the two cases

Both changes are breaking, the one for windows clients, the other for linux clients.

Reply:
Of those two I'd go with the first option.

Reply:
Ok, I'll change it that way.